Update Singer-specific pages to be Meltano-neutral

tayloramurphy opened the pull request:
This PR updates the Singer pages on the develop branch to be more like what currently exists.

TODO:

- [ ] Update alternate variant URLs

tayloramurphy commented:
@DouweM I could use a review on this. Locally I'm able to build it and the default variants work just like you'd expect. On this build though I'm seeing the wrong variant as the default page even though it's listed in the variants list as `(default)`. https://deploy-preview-541--meltano-hub.netlify.app/taps/auth0

Maybe it's something with my Ruby version locally? I'm using 2.7.

Otherwise I'm pretty happy with this PR - it's actually better than what's currently on the Hub! https://hub.meltano.com/taps/auth0

tayloramurphy commented:
I've updated Ruby to 3.1.2 and I see the same behavior currently on develop. Looks like it's perhaps a sorting thing?

tayloramurphy commented:
Found it! 🎉  6f0045e31ef928862e8af4b877ee5d1687a94f0b

tayloramurphy commented:
@DouweM @alexmarple this is ready for review. 

A nice side by side comparison to see current state vs what's in this PR are the Auth0 pages listed in https://github.com/meltano/hub/pull/541#issuecomment-1146514081

We're still building both sets of pages from a single HTML file which is fine I think for the launch and will change presumably with the Gridsome Redesign.

DouweM commented:
@tayloramurphy Comparing them side by side is pretty handy, I spotted a few differences already that I think we should fix:
- [ ] Page title says "Meltano tap" instead of "Singer tap"
- [ ] Navbar says "MeltanoHub" instead of "MeltanoHub for Singer"
- [ ] Description says "the Auth0" without a link instead of "Auth0" with a link
- [ ] Repository list lacks "Last Updated At"
- [ ] "Meltano Stats" are missing
- [ ] "Edit this page on GitLab" is missing

Except for the first 2, all of these changes would be good on the Meltano-specific pages as well. I'm actually not sure why these details went missing in the first place, if they're currently live on hub.meltano.com?

I also suggest changing the "Usage with Meltano" section to make the point of Meltano a little more clear:

> [Meltano](https://meltano.com) helps you manage your [configuration](https://docs.meltano.com/guide/configuration), [incremental replication](https://docs.meltano.com/guide/integration#incremental-replication-state), and [scheduled pipelines](https://docs.meltano.com/guide/orchestration#create-a-schedule). 
> 
> View the [Meltano-specific `tap-auth0` instructions](/extractors/auth0#getting-started) to learn more.

Note that the second link includes `#getting-started` so that the user is directed to the new section, and not a page that at first glance looks exactly like the page they just saw.

tayloramurphy commented:
> Page title says "Meltano tap" instead of "Singer tap"
> Navbar says "MeltanoHub" instead of "MeltanoHub for Singer"

I'm going to pop a separate issue to track that. That happened b/c of the consolidation of the Meltano and Singer pages into one so I'll need to spend more time on fixing that, but I'd like to keep it separate from the pure content part if that's ok. https://github.com/meltano/hub/issues/548

> Description says "the Auth0" without a link instead of "Auth0" with a link

Fixed

> Repository list lacks "Last Updated At"
> "Meltano Stats" are missing

This is because the local (and presumably Netlify) versions of the Hub are using a smaller static file to pull in metrics. The full deploy process gets everything from S3 (cc @pnadolny13 if I've mispoken here). If you look at https://deploy-preview-541--meltano-hub.netlify.app/taps/slack you'll see the metrics. 

> "Edit this page on GitLab" is missing

Fixed and changed to GitHub!

> I also suggest changing the "Usage with Meltano" section

Done!

DouweM commented on _includes/setting.html:
```diff
@@ -7,6 +7,8 @@ <h3 id="{{ setting.name | slugify }}">
   {% endif %}
 </h3>
 
+{% if is_singer == false %}
```
Is `is_singer` actually available here? As far as I know, variables have to be passed into included partials explicitly and then referenced as `include.<var>` as in line 1 of this file.

tayloramurphy replied:
If you compare the settings on https://deploy-preview-541--meltano-hub.netlify.app/extractors/auth0 and https://deploy-preview-541--meltano-hub.netlify.app/taps/auth0 it does seem to work. 🤷

DouweM commented on _layouts/meltano_plugin.html:
```diff
@@ -9,6 +9,44 @@
 {% if plugin.metrics %} {% assign metrics = plugin.metrics %}{% endif %}
 
+<!-- To determine if the current plugin is pure Singer or not -->
+{% assign is_singer = false %}
```
We may be better off moving this logic to https://github.com/meltano/hub/blob/main/_plugins/meltano_plugin_page_generator.rb so that we can also use it to toggle the page title and header (https://github.com/meltano/hub/issues/548).

If we add it to `@data` there, it becomes available as `page.is_singer` here, and likely in `_includes/setting.html` as well -- although we may still need to pass it there explicitly.

tayloramurphy replied:
@DouweM I was able to move the logic and update everything. Much easier to do it in the Ruby script!

I'm going to open another PR targeting this branch that will work to fix the other plugin pages. I've got a few commits ready but will have to wait until tomorrow to continue it.

DouweM reviewed (changes requested):
One last potential issue and a workaround that may also help with #548.

I also just noticed that the page descriptions on https://deploy-preview-541--meltano-hub.netlify.app/taps/auth0 is wrong:
> The open source Meltano loader sends data into Auth0 after it was pulled from a source using an extractor.

https://deploy-preview-541--meltano-hub.netlify.app/extractors/auth0 is right:
> The open source Meltano extractor pulls data from Auth0 that can then be sent to a destination using a loader.

We may only have `if extractor; else` logic, meaning that anything not an extractor gets the loader text.

pnadolny13 reviewed (approved):
@tayloramurphy thanks for the quick fixes! Looks good to me.

DouweM reviewed (changes requested):
- [ ] The page description incorrectly describes anything other than extractors as loaders

tayloramurphy commented on _layouts/meltano_plugin.html:
```diff
@@ -5,41 +5,32 @@
 {% endcapture %}
 {% if plugin.metrics %} {% assign metrics = plugin.metrics %}{% endif %}
 
 <!-- To determine if the current plugin is pure Singer or not -->
```
Fixed

tayloramurphy commented on _layouts/default.html:
```diff
@@ -3,8 +3,13 @@
-    include header.html %} {% endif %}
+    {% if page.header %} 
+      {% include header_{{page.header}}.html %} 
+    {% elsif page.is_singer == true%} 
```
Per our DMs, going to leave as is and merge.

tayloramurphy commented:
> The page description incorrectly describes anything other than extractors as loaders

Addressing this in #550
